golem-test-framework/src/components/rdb/k8s_postgres.rs

```diff
@@ -0,0 +1,164 @@
+use async_dropper_simple::{AsyncDrop, AsyncDropper};
+use async_scoped::TokioScope;
+use k8s_openapi::api::core::v1::{Pod, Service};
+use kube::api::PostParams;
+use kube::{Api, Client};
+use serde_json::json;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::info;
+
+pub struct K8sPostgresRdb {
+    _namespace: K8sNamespace,
+    pod: Arc<Mutex<K8sPod>>,
+    service: Arc<Mutex<K8sService>>,
+    routing: Arc<Mutex<K8sRouting>>,
+    host: String,
+    port: u16,
+}
+
+impl K8sPostgresRdb {
+    pub async fn new(namespace: &K8sNamespace, routing_type: &K8sRoutingType) -> Self {
+        info!("Creating Postgres pod");
+
+        let pods: Api<Pod> = Api::namespaced(
+            Client::try_default()
+                .await
+                .expect("Failed to create K8s client"),
+            &namespace.0,
+        );
+        let services: Api<Service> = Api::namespaced(
+            Client::try_default()
+                .await
+                .expect("Failed to create K8s client"),
+            &namespace.0,
+        );
+
+        let pod: Pod = serde_json::from_value(json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": {
+                "name": "golem-postgres",
+                "labels": {
+                    "app": "golem-postgres",
+                    "app-group": "golem"
+                },
+            },
+            "spec": {
+                "ports": [{
+                    "port": 5432,
+                    "protocol": "TCP"
+                }],
+                "containers": [{
+                    "name": "postgres",
+                    "image": "postgres:12",
+                    "env": [
+                        {"name": "POSTGRES_DB", "value": "postgres"},
+                        {"name": "POSTGRES_USER", "value": "postgres"},
+                        {"name": "POSTGRES_PASSWORD", "value": "postgres"}
+                    ]
+                }]
+            }
+        }))
+        .expect("Failed to deserialize pod definition");
+
+        let pp = PostParams::default();
+
+        let _res_pod = pods.create(&pp, &pod).await.expect("Failed to create pod");
+        let managed_pod = AsyncDropper::new(ManagedPod::new("golem-postgres", namespace));
+
+        let service: Service = serde_json::from_value(json!({
+            "apiVersion": "v1",
+            "kind": "Service",
+            "metadata": {
+                "name": "golem-postgres",
+                "labels": {
+                    "app": "golem-postgres",
+                    "app-group": "golem"
+                },
+            },
+            "spec": {
+                "ports": [{
+                    "port": 5432,
+                    "protocol": "TCP"
+                }],
+                "selector": { "app": "golem-postgres" },
+                "type": "LoadBalancer"
+            }
+        }))
+        .expect("Failed to deserialize service description");
+
+        let _res_srv = services
```

Comment:
There is no need for explicit name here. There is no `Drop`. Explicit name might be misleading - it keeps variable alive till the end of the method (i.e. prevents immediate `Drop` call).